### PR TITLE
[parquet-sdk-migration] add a logic to determine the starting version for parquet processor

### PR DESCRIPTION
**Pull request by yuunlimm**
https://www.notion.so/aptoslabs/Update-run_processor-Function-12d8b846eb7280268d65c0d92d2e328a?pvs=4

### CODE REVIEW

**Comment by yuunlimm** [comment]
* **#580** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-indexer-processors/580?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @yuunlimm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: rust/sdk-processor/src/config/processor_config.rs [hunk 1]
```diff
@@ -55,6 +55,28 @@ impl ProcessorConfig {
...
+    /// Get the Vec of table names for parquet processors only.
+    /// This is a convenience method to map the table names to include the processor name as a prefix, which
+    /// is useful for querying the status from the processor status table in the database.
+    pub fn get_table_names(&self) -> Option<Vec<String>> {
+        match self {
+            ProcessorConfig::ParquetDefaultProcessor(config) => {
+                // Get the processor name as a prefix
+                let prefix = self.name();
+                // Use the tables from the config and map them to include the prefix
+                println!("config.tables: {:?}", prefix);
```
Comment (dermanyang) [inline rust/sdk-processor/src/config/processor_config.rs:68]: leftover print statemenet?

File: rust/sdk-processor/src/config/processor_config.rs [hunk 2]
```diff
@@ -55,6 +55,28 @@ impl ProcessorConfig {
     pub fn name(&self) -> &'static str {
         self.into()
     }
+
+    /// Get the Vec of table names for parquet processors only.
```
Comment (dermanyang) [inline rust/sdk-processor/src/config/processor_config.rs:59]: nit: add newline after first sentence for nicer [docstring parsing](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#documenting-components)

File: rust/sdk-processor/src/utils/starting_version.rs [hunk 3]
```diff
@@ -42,6 +42,77 @@ pub async fn get_starting_version(
     ))
 }
 
+pub async fn get_minimum_last_success_version(
```
Comment (dermanyang) [inline rust/sdk-processor/src/utils/starting_version.rs:45]: Let's add a docstring here + add append `parquet` to the end of the function name oslt. What do you think? I'm worried that colocating these functions w/ the other postgres ones may be confusing.

File: rust/sdk-processor/src/utils/starting_version.rs [hunk 4]
```diff
@@ -42,6 +42,77 @@ pub async fn get_starting_version(
...
+    // If nothing checkpointed, return the `starting_version` from the config, or 0 if not set.
+    Ok(min_processed_version.unwrap_or(
+        indexer_processor_config
+            .transaction_stream_config
+            .starting_version
+            .unwrap_or(0),
+    ))
+}
+
+async fn get_processed_version_from_db(
```
Comment (dermanyang) [inline rust/sdk-processor/src/utils/starting_version.rs:63]: `get_processed_version_from_db` -> `get_min_processed_version_from_db` or move the min to the calling function

**Review by dermanyang — APPROVED** [review]
thanks for changes. lgtm

**Pull request merged**